### PR TITLE
Refactored for bool  - Fixes for wrong links in emails

### PR DESCRIPTION
**Pull request by datdamnzotz**

### CODE REVIEW

**Review by Demonsthere — APPROVED** [review]

**Comment by Demonsthere** [comment]
Yup, now the vars are nice :D

File: src/clj/orcpub/email.clj [hunk 1]
```diff
@@ -30,15 +35,15 @@
    :host (environ/env :email-server-url)
    :port (Integer/parseInt (environ/env :email-server-port))})
 
-(defn send-verification-email [base-url {:keys [email username first-and-last-name]} verification-key]
+(defn send-verification-email [{:keys [email username first-and-last-name]} verification-key]
```
Comment (MeanderingCode) [inline src/clj/orcpub/email.clj:38]: You should not pull out base-url.

This is defining it as an argument to the function, which is called in the router src/clj/orcpub/routes.clj There, the request headers are being used to determine what the hostname is.

You should remove your environment variables, return this argument, and we should look into why it is getting the wrong request headers. I suspect it is either not a problem, just an artifact of some setup, or due to the nginx reverse proxy config munging the request headers.

File: src/clj/orcpub/email.clj [hunk 2]
```diff
@@ -65,20 +70,22 @@
...
-  (postal/send-message (email-cfg)
-                       {:from "OrcPub Errors <no-reply@orcpub.com>"
-                        :to "redorc@orcpub.com"
-                        :subject "Exception"
-                        :body [{:type "text/plain"
-                                :content (let [writer (java.io.StringWriter.)]
-                                           (do (clojure.pprint/pprint (:request context) writer)
-                                               (clojure.pprint/pprint (or (ex-data exception) exception) writer)
-                                               (str writer)))}]}))
+  (if ( = true (environ/env :email-errors-enabled))
```
Comment (MeanderingCode) [inline src/clj/orcpub/email.clj:81]: Check here for environ/env :email-errors-to having a value.

File: src/clj/orcpub/email.clj [hunk 3]
```diff
@@ -4,6 +4,11 @@
             [environ.core :as environ]
             [orcpub.route-map :as routes]))
 
+(defn get-http-protocol [] ()
```
Comment (MeanderingCode) [inline src/clj/orcpub/email.clj:7]: Unnecessary if we are keeping the `base-url` argument to the `send-verification-email` function

File: docker-compose.yaml [hunk 4]
```diff
@@ -9,11 +9,19 @@ services:
...
+      SSL_ENABLED: true
       PORT: 8890
       EMAIL_SERVER_URL: ''
       EMAIL_ACCESS_KEY: ''
       EMAIL_SECRET_KEY: ''
       EMAIL_SERVER_PORT: 587
+      # Email Errors? true/false
+      EMAIL_ERRORS_ENABLED: false
+      # Email address to send errors to
+      EMAIL_ERRORS_TO: ''
```
Comment (MeanderingCode) [inline docker-compose.yaml:24]: Given the other review comments, this will be the only new envvar we need.

**Review by MeanderingCode — CHANGES_REQUESTED** [review]
As mentioned in another comment, I think there is a different issue at play. We should test the dockerized setup and determine what `HOST` header the application receives when it is behind the nginx reverse proxy, and fix that configuration.

I love the conditional "if an address for errors is set, then email the errors" feature.

**Comment by datdamnzotz** [comment]
Issue lies in the base-url, which when behind the proxy we need to be the url it sees.
I'll close this and take another hard look

**Pull request closed without merging**